lib/v2/screens/dashboard/interactor/mappers/token_balances_state_mapper.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:seeds/v2/datasource/remote/model/balance_model.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/token_balance_view_model.dart';
+import 'package:seeds/v2/datasource/remote/model/token_model.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/dashboard/interactor/viewmodels/token_balances_state.dart';
+
+class TokenBalancesStateMapper {
+
+  TokenBalancesState mapResultToState(TokenBalancesState currentState, List<TokenModel> tokens, List<Result> results) {
+
+    assert(tokens.length == results.length, "invalid results");
```

Comment:
is this going to crash the app?

Reply:
if the `==`.    fails

Reply:
No crash, asssert are ignored in production.

Reply:
Well these lists need to be the same length. That's all.